### PR TITLE
minion-ffi: add timeout in tests

### PR DESCRIPTION
**Pull request by sleirsgoevy**

### CODE REVIEW

File: src/minion-ffi/tests/runner.c [hunk 1]
```diff
@@ -177,15 +187,22 @@ int main(int argc, const char** argv) {
...
+            fprintf(stderr, "test `%s`: ", tests[i].name);
+            if(is_timeout)
+                fprintf(stderr, "timeout");
+            else
+                fprintf(stderr, "output differs");
             fprintf(stderr,
-                    "test `%s`: output differs:\nActual output (len %lld):\n",
-                    tests[i].name, (long long) output_sz);
+                    ":\nActual output (len %lld):\n",
+                    (long long) output_sz);
```
Comment (MikailBag) [inline src/minion-ffi/tests/runner.c]: Do you kill child?

File: src/minion-ffi/tests/runner.c [hunk 2]
```diff
@@ -117,11 +117,21 @@ int test_main(int argc, char const* const* argv) {
     return 179;
 }
 
-size_t read_all(int fd, const char** buf_p) {
+size_t read_all(int fd, const char** buf_p, int* is_timeout) {
+    *is_timeout = 0;
+    struct timeval timeout = {20, 0};
```
Comment (MikailBag) [inline src/minion-ffi/tests/runner.c]: I think it should be derived from test timeout, e.g.
test_timeout*1.5

File: src/minion-ffi/tests/tests.c [hunk 3]
```diff
@@ -55,5 +55,5 @@ const struct test tests[] = {
     {"return1", test_return_1, "exit code 1\n", 1, 2},
     {"ok", test_ok, "exit code 0\n", 1, 2},
     {"consume_memory", test_consume_memory, "exit code -9\n", 10, 25},
-    {"wait_timeout", test_il, "Wait timed out\n", 1, 10},
+    //{"wait_timeout", test_il, "Wait timed out\n", 1, 10},
```
[the remark this answers is not in the document]
Comment (sleirsgoevy) [reply src/minion-ffi/tests/tests.c]: It is broken, so no, it shouldn't.

File: src/minion-ffi/tests/runner.c (hunk 1 above)
Comment (sleirsgoevy) [reply src/minion-ffi/tests/runner.c]: Very interesting question... probably no

**Pull request closed without merging**